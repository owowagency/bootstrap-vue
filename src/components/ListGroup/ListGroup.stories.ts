import ListGroup from '.';
import ListGroupItem from '@/components/ListGroupItem';
import {variants} from '@/composables/useVariant';

export default {
    title: 'Components/List group',
    component: ListGroup,
    argTypes: {},
};

const template = (args) => ({
    components: {ListGroup, ListGroupItem},
    setup: () => ({args}),
    template: `
        <ListGroup v-bind="args">
            <ListGroupItem active>An active item</ListGroupItem>
            <ListGroupItem>A second item</ListGroupItem>
            <ListGroupItem disabled>A disabled item</ListGroupItem>
            <ListGroupItem>A fourth item</ListGroupItem>
            <!-- An action has 100% width. This does not work well with horizontal -->
            <ListGroupItem :tag="args.horizontal ? 'div' : 'a'">An action item</ListGroupItem>
            <ListGroupItem>And a sixth one</ListGroupItem>
        </ListGroup>
    `,
});

export const Default = template.bind({});

export const Flush = template.bind({});

Flush.args = {
    flush: true,
};

export const Numbered = template.bind({});

Numbered.args = {
    numbered: true,
};

export const Horizontal = template.bind({});

Horizontal.args = {
    horizontal: true,
};

export const Variants = (args) => ({
    components: {ListGroup, ListGroupItem},
    setup: () => ({args, variants}),
    template: `
        <ListGroup v-bind="args">
            <ListGroupItem
                v-for="v of variants"
                :variant="v"
                tag="button"
            >
                {{ v }}
            </ListGroupItem>
        </ListGroup>
    `,
});
