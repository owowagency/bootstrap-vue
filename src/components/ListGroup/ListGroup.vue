<template>
    <Component
        :is="tag"
        class="list-group"
        :class="classes"
    >
        <!-- @slot Displays the content of the list group -->
        <slot />
    </Component>
</template>

<script lang="ts">
import useFlush, {flushProps} from '@/composables/useFlush';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...flushProps,
    /**
     * Defines if the list should be displayed horizontally
     */
    horizontal: {
        type: Boolean,
        default: false,
    },
    /**
     * Defines if the list should be numbered
     */
    numbered: {
        type: Boolean,
        default: false,
    },
});

const tag = computed(() => props.numbered ? 'ol' : 'ul');

const {classes} = useClasses(computed(() => [
    useFlush(props.flush, 'list-group-{0}').flushClass.value,
    props.horizontal ? 'list-group-horizontal' : undefined,
    props.numbered ? 'list-group-numbered' : undefined,
]));
</script>

<docs>
```vue
<ListGroup>
    <ListGroupItem>...</ListGroupItem>
    <ListGroupItem>...</ListGroupItem>
    <ListGroupItem>...</ListGroupItem>
</ListGroup>
```

## Imported Props

| Prop name | Description                            | Type    | Values | Default |
| --------- | -------------------------------------- | ------- | ------ | ------- |
| flush     | Use the flush variant of the component | boolean | -      | false   |
</docs>
