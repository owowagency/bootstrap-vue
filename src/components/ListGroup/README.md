# ListGroup

```vue
<ListGroup>
    <ListGroupItem>...</ListGroupItem>
    <ListGroupItem>...</ListGroupItem>
    <ListGroupItem>...</ListGroupItem>
</ListGroup>
```

## Imported Props

| Prop name | Description                            | Type    | Values | Default |
| --------- | -------------------------------------- | ------- | ------ | ------- |
| flush     | Use the flush variant of the component | boolean | -      | false   |

## Props

| Prop name  | Description                                          | Type    | Values | Default |
| ---------- | ---------------------------------------------------- | ------- | ------ | ------- |
| horizontal | Defines if the list should be displayed horizontally | boolean | -      | false   |
| numbered   | Defines if the list should be numbered               | boolean | -      | false   |

## Slots

| Name    | Description                            | Bindings |
| ------- | -------------------------------------- | -------- |
| default | Displays the content of the list group |          |
