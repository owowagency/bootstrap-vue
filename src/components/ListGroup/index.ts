import ListGroup from './ListGroup.vue';

export default ListGroup;
