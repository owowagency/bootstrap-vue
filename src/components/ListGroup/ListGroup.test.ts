import ListGroup from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(ListGroup);

    componentSlotRenderTest(ListGroup);

    componentWrapperClassTest(ListGroup, {flush: true}, 'list-group-flush');

    componentWrapperClassTest(ListGroup, {horizontal: true}, 'list-group-horizontal');

    componentWrapperClassTest(ListGroup, {numbered: true}, 'list-group-numbered');

    componentWrapperTagTest(ListGroup, {props: {tag: 'ul', numbered: false}});

    componentWrapperTagTest(ListGroup, {props: {tag: 'ol', numbered: true}});
});


