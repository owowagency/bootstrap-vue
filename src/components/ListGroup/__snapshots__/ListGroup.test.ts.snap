// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<ul class=\\"list-group\\">
  <!-- @slot Displays the content of the list group -->
</ul>"
`;
